core/src/main/java/de/danoeh/antennapod/core/service/download/NewEpisodesNotification.java

```diff
@@ -0,0 +1,84 @@
+package de.danoeh.antennapod.core.service.download;
+
+import android.app.Notification;
+import android.app.PendingIntent;
+import android.content.ComponentName;
+import android.content.Context;
+import android.content.Intent;
+import android.content.res.Resources;
+
+import androidx.core.app.NotificationCompat;
+import androidx.core.app.NotificationManagerCompat;
+
+import de.danoeh.antennapod.core.R;
+import de.danoeh.antennapod.core.feed.Feed;
+import de.danoeh.antennapod.core.feed.FeedPreferences;
+import de.danoeh.antennapod.core.storage.DBReader;
+import de.danoeh.antennapod.core.storage.PodDBAdapter;
+import de.danoeh.antennapod.core.util.gui.NotificationUtils;
+
+public class NewEpisodesNotification {
+    static final String GROUP_KEY = "de.danoeh.antennapod.EPISODES";
+
+    private final int lastEpisodeCount;
+    private final boolean dontShowNotification;
+
+    public NewEpisodesNotification(Long feedId) {
+        Feed feed = DBReader.getFeed(feedId);
+
+        FeedPreferences prefs = feed.getPreferences();
+        if (!prefs.getKeepUpdated() || !prefs.getShowEpisodeNotification()) {
+            dontShowNotification = true;
+            lastEpisodeCount = -1;
+            return;
+        }
+
+        lastEpisodeCount = PodDBAdapter.getInstance().getFeedCounters(feedId).get(feedId);
```

Comment:
`getFeedCounters` loads a different counter depending on user settings. Please add another argument to that method and call `getFeedCounters(UserPreferences.FEED_COUNTER_SHOW_NEW, feedId)` here. The only other usage of that method can then pass the user setting.

I mean something like this in PodDBAdapter: 

```
public final LongIntMap getFeedCounters(int setting, long... feedIds) {
```